### PR TITLE
Package `kmarshal`

### PR DESCRIPTION
**Pull request by FUDCo**
This moves the `kmarshal` module into its own package and then updates everybody who imported it to use _that_.

This also eliminates the duplicate implementations (one in `swingset-liveslots` and one in `SwingSet`) that needed to be maintained in parallel.

My original intent had been to fold this into the forthcoming refactor that moves around the benchmark tooling and bootstrap test support code so that the modularity of that stuff makes sense, but this repackaging touched so many files that it seemed to me to be prudent to review it as its own thing.  Note that although it does touch a _lot_ of files, almost all the changes are superficial.

This is in support of #8327

### CODE REVIEW

**Comment by mhofman** [comment]
> This also eliminates the duplicate implementations (one in `swingset-liveslots` and one in `SwingSet`) that needed to be maintained in parallel.

FYI I am slightly wary of shared dependencies between liveslots and the kernel as it gives us a false sense of ability to atomically make changes. I think we should only do that if we have strong tests of the external interface of the package, make sure it stays 100% compatible if the swingset side upgrades but not the liveslots side.

Edit: that said, if we have such strong tests, an independent package like this one which basically describes the protocol between kernel and vats is extremely valuable as it helps us be more formal about defining what the interface looks like.

**Comment by FUDCo** [comment]
> FYI I am slightly wary of shared dependencies between liveslots and the kernel as it gives us a false sense of ability to atomically make changes. I think we should only do that if we have strong tests of the external interface of the package, make sure it stays 100% compatible if the swingset side upgrades but not the liveslots side.

Fair point, but I think the risk of divergence under maintenance of two copies of the identical file in two different places is a more immediate practical concern.  Note that the use in liveslots is 100% in tests.

File: packages/kmarshal/test/test-basic.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import test from 'ava';
+
+import { kser, kunser, krefOf } from '../src/kmarshal.js';
+
+test('sanity check', t => {
+  const serUndefined = kser(undefined);
+  t.deepEqual(serUndefined, {
```
Comment (gibson042) [inline packages/kmarshal/test/test-basic.js]: Hmm, I didn't realize we were misusing "ser[ialize]" to refer to non-serial data structures.

File: packages/kmarshal/test/test-basic.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    body: '#"#undefined"',
+    slots: [],
+  });
+  t.is(kunser(serUndefined), undefined);
+
+  const serNumber = kser(47);
+  t.deepEqual(serNumber, {
+    body: '#47',
```
Comment (gibson042) [inline packages/kmarshal/test/test-basic.js]: Do we actually care that `kser` emits CapData with a smallcaps body? Do we care that it emits CapData? I think we can get by with less, e.g.
```js
const assertRoundTrip = (t, val, toComparable = v => v) => {
  const ser = kser(val);
  t.not(ser, val);
  const unser = kunser(ser);
  t.is(toComparable(unser), toComparable(val));
};
assertRoundTrip(t, undefined);
assertRoundTrip(t, 47);
assertRoundTrip(t, -2n);
assertRoundTrip(t, 'foo');
```

File: packages/kmarshal/test/test-basic.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  t.is(kunser(serNumber), 47);
+
+  const serString = kser('foo');
+  t.deepEqual(serString, {
+    body: '#"foo"',
+    slots: [],
+  });
+  t.is(kunser(serString), 'foo');
+
+  const remotable = kunser({ body: '#"$0"', slots: ['ko47'] });
```
Comment (gibson042) [inline packages/kmarshal/test/test-basic.js]:
```suggestion
  const remotable = kslot('ko47');
  t.is(krefOf(remotable), 'ko47');
  assertRoundTrip(t, remotable, krefOf);
```

**Review by gibson042 — APPROVED** [review]
I would have preferred separation of substantive work and mechanical changes into independent commits, but this wasn't difficult to get through as one. Thanks for making a dedicated PR!

File: packages/kmarshal/test/test-basic.js (hunk 2 above)
Comment (FUDCo) [reply packages/kmarshal/test/test-basic.js]: That's lovely.

**Pull request merged**